### PR TITLE
Fix implicit fallthrough warning on GCC 7.x

### PR DESCRIPTION
**Pull request by afirago**
Add fallthrough comments to keep GCC >= 7.x happy.

Signed-off-by: Alexey Firago <alexey_firago@mentor.com>

### CODE REVIEW

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by afirago** [comment]
CLA is updated

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

File: src/core/lib/support/murmur_hash.c [hunk 1]
```diff
@@ -61,15 +61,15 @@ uint32_t gpr_murmur_hash3(const void *key, size_t len, uint32_t seed) {
...
-    case 1:
-      k1 ^= tail[0];
-      k1 *= c1;
-      k1 = ROTL32(k1, 15);
-      k1 *= c2;
-      h1 ^= k1;
+      k1 ^= ((uint32_t)tail[2]) << 16; //-fallthrough
+    case 2: //-fallthrough
+      k1 ^= ((uint32_t)tail[1]) << 8; //-fallthrough
+    case 1: //-fallthrough
```
Comment (ctiller) [inline src/core/lib/support/murmur_hash.c]: surely this doesn't need to happen on every line of the case stanza?

Comment (afirago) [inline src/core/lib/support/murmur_hash.c]: @ctiller thanks for checking. Fixed it to required minimum of comments.

Comment (ctiller) [inline src/core/lib/support/murmur_hash.c]: Would it read better as:
```c
    case 2:
       k1 ^= ((uint32_t)tail[1]) << 8; 
       //-fallthrough
```

Comment (afirago) [inline src/core/lib/support/murmur_hash.c]: @ctiller done!

**Comment by ctiller** [comment]
Jenkins: this is ok to test

**Comment by ctiller** [comment]
Care to make one last change to keep our clang-format checks consistent:
```
   switch (len & 3) {
     case 3:
       k1 ^= ((uint32_t)tail[2]) << 16;
-      //-fallthrough
+    //-fallthrough
     case 2:
       k1 ^= ((uint32_t)tail[1]) << 8;
-      //-fallthrough
+    //-fallthrough
     case 1:
       k1 ^= tail[0];
       k1 *= c1;
```

**Comment by ctiller** [comment]
(I think clang-format is wrong here, but it's better to keep it happy)

**Comment by afirago** [comment]
@ctiller  
Should I also change "//-fallthrough" comments to /* fallthrough */ to be consistent with the coding style?

Also I've found several similar issues in other parts of the source tree. Should I add more commits to this PR or close this one and open new with one commit covering all "fallthrough" warnings?

**Comment by ctiller** [comment]
/* fallthrough */ would be better for us, although I'm not terribly fussy.
Feel free to append this PR with more fixes - I can test them all in a
batch.

We have a wrapper script, tools/distrib/clang_format_code.sh that does all
the code formatting in a consistent way (it'll need docker to be installed)
if that's helpful for getting it all polished.

**Comment by afirago** [comment]
Updates:
* Fixed indentation as requested
* Changed "//-fallthrough" to preferred "/* fallthrough */ 
* Add more fixes to the commit for the same issue across the source tree. Updated commit and PR name

**Pull request merged**